docs/okeps/okep-5377-extend-udn-to-support-multiple-cluster-subnets-in-layer3-topology.md

```diff
@@ -0,0 +1,284 @@
+```
+$ ovn-nbctl lr-policy-list  udn_udn.primary.layer3_ovn_cluster_router
+Routing Policies
+      ...
+      102   ip4.src == 10.1.0.0/16 && ip4.dst == 10.1.0.0/16    allow
+      102   ip4.src == 10.1.0.0/16 && ip4.dst == 10.2.0.0/16    allow
+      102   ip4.src == 10.2.0.0/16 && ip4.dst == 10.2.0.0/16    allow
+      102   ip4.src == 10.2.0.0/16 && ip4.dst == 10.1.0.0/16    allow
+```
+
+
+### API Validations
+
+The `subnets` field in the current CRD has the following validation rules:
+```
+	// +kubebuilder:validation:MaxItems=2
+	// +kubebuilder:validation:XValidation:rule="size(self) != 2 || !isCIDR(self[0].cidr) || !isCIDR(self[1].cidr) || cidr(self[0].cidr).ip().family() != cidr(self[1].cidr).ip().family()", message="When 2 CIDRs are set, they must be from different IP families"
+	Subnets []Layer3Subnet `json:"subnets,omitempty"`
+```
+
+* `MaxItems` is required to keep CEL evaluation costs below threshold. It has to be
+  retained but adjusted to a reasonable value (the max value is 400 for the new rules defined below).
+* The `XValidation:rule` must be removed to allow more than one subnet per IP family.
+
+New validation logic should be added to enforce the following constraints on `layer3.subnets`:
+* Prevent removal or modification of existing subnets:
+  ```yaml
+    - message: Removing existing subnets is not allowed
+      rule: '!has(oldSelf.subnets) || oldSelf.subnets.all(old, self.subnets.exists(new,
+        new.cidr == old.cidr))'
+  ```
+* Prevent overlapping or nested subnets:
+  ```yaml
+    - message: Subnets must not overlap or contain each other
+      rule: '!has(self.subnets) || self.subnets.size() == 1 || !self.subnets.exists(i,
+        self.subnets.exists(j, i != j && cidr(i.cidr).containsCIDR(j.cidr)))'
+  ```
+* Ensure all subnets use the same hostSubnet size:
+  ```yaml
+    - message: All subnets must use the same hostSubnet value
+      rule: '!has(self.subnets) || self.subnets.size() == 1 || self.subnets.all(i,
+        i.hostSubnet == self.subnets[0].hostSubnet)'
+  ```
+
+Additionally, the validation rule that makes the entire UDN `spec` immutable must be removed:
+```
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Spec is immutable"
+	Spec UserDefinedNetworkSpec `json:"spec"`
+```
+Instead, immutability should be enforced at the sub-field level to preserve immutability where
+needed.
+
+
+### Testing Details
+
+* **Unit Tests:** Extend subnet allocator tests to cover multiple subnets and dynamic expansion scenarios.
+* **E2E Tests:** Test UDN behavior when new subnets are appended, and verify that nodes receive
+  allocations from the correct ranges.
+* **API Tests:** Validate CRD updates for append-only behavior and ensure invalid changes (e.g., removal
+  or modification of existing subnets) are correctly rejected.
+
+
+### Documentation Details
+
+* Update `mkdocs.yml` with the OKEP reference.
```

Comment:
Documentation details isn't updating the mkdocs with the OKEP reference. It should be something like "UDN feature documentation will be updated to declare support for multiple subnets"